### ISSUE TITLE
Does changeset react to changes on the underlying model?

### ISSUE THREAD

**Issue by lougreenwood** [poteto/ember-changeset#340]
I have a model for which I am setting some default relationship values.

Because Ember Data has no obvious API for setting default relationship values, I'm using the `DS.model.ready()` hook to set the relationships.

The problem is that ready does not fire immediately after a model is created. So for example, in this code, the model `ready()` is not fired before the changeset is created:

```
        const model = this._createModel(modelName);
        const changeset = this._createChangeset(model, changesetValidations);

        // ....
        // model.ready() is fired somewhere down here
```

This leads to a problem where the changeset is out of sync with the model because the default value was set on the model **after** the changeset is created but the changes knows nothing about this.

So... I'm wondering 2 things:

* Does changeset have any mechanism to detect changes to the underlying model.
* Does changeset have any mechanism to merge model data back into a changeset?
  * For this I assume that a workaround might be to create a 2nd changeset after `model.ready` fires and then something like `newChangeset = changesetOld.merge(changesetAfterReady)`

**Comment by snewcomer** [comment]
This may be a similar flavor of a problem to https://github.com/poteto/ember-changeset/issues/339.

Is `model.ready` promise based?  Otherwise, I'm not sure how we can tell inside of changeset.

**Comment by nickschot** [comment]
AFAIK changeset itself does not detect changes to the underlying model. I am currently manually creating a new changeset and merging existing unsaved changes into that whenever the model is updated (in the few places where I know the model can be updated based on user interaction outside of the changeset).

**Comment by snewcomer** [comment]
That is probably the best route since coupling behaviours too tightly may be outside the scope of this addon.

**Comment by lougreenwood** [comment]
So I managed to find a solution where I can set default relationships in ED by overriding the init hook (seems this isn't documented, so I initially went with the `ready` hook).

The init hook means that all default rels are set before the model is passed to EC, which essentially circumvents the problem I initially described.

So yeah, I think we can close this issue since I managed to achieve a more robust solution.

Thanks!
